grammar.ebnf

```diff
@@ -36,11 +38,40 @@ Statement = "{", {Statement}, "}"
 */
 LValue = Identifier, [ ".", Identifier ];
 
+Expression = ExpressionTerm 
+           | ParenthesisExpression
+           | FunctionCall
+           | NotExpression
+           | ArrayLookupExpression
+           | BinaryExpression
+
+BinOp = "+" | "-" | "*" | "/" | "%" | "&&" | "||" | "<" | "=<"| ">" | ">="| "=="| "!=";
+
+ParenthesisExpression = "(", Expression, ")";
+
+FunctionCall = Identifier, "(", FuncCallArgList, ")";
+
 FuncCallArgList = [ Expression, {",", Expression} ];
 
+
+BinaryExpression = ExpressionTerm, BinOp, ExpressionTerm;
+
+NotExpression = "!", Expression;
+
+ArrayLookupExpression = ExpressionTerm, "[", ExpressionTerm, "]";
+
+ExpressionTerm = Integer
+               | Identifier 
```

Comment:
We should let `LValue` here.

Reply:
You mean, add it to the ExpressionTerm rule?

Reply:
No need to add it. You can replace it with `Identifier` because an `LValue` can be an `Identifier`.